tests/testthat/test-thumbnail.R

```diff
@@ -0,0 +1,125 @@
+mock_dirs <- c(
+  "unversioned" = "2024.08.0",
+  "v1" = "2024.09.0"
+)
+
+# The mocked content items are as follows:
+# - 01234567:
+#   - Returns 200 and a tiny jpeg when getting the thumbnail
+#   - Returns 204 (success) when setting and deleting
+# - 12345678: Returns 204 when getting the thumbnail (no thumbnail)
+# - 23456789: Returns 404 for all endpoints
+# - non-connect/missing-image/image.R: a 404 response
+
+for (api_ver in names(mock_dirs)) {
+  mock_dir <- mock_dirs[[api_ver]]
+  with_mock_dir(mock_dir, {
+    with_mock_api({
+      client <- connect(server = "https://connect.example", api_key = "fake")
+      test_that(glue::glue("get_thumbnail() gets the thumbnail ({api_ver} - {mock_dir})"), {
+        item <- content_item(client, "01234567")
+        
+        # User-specified path
+        user_path <- tempfile("thumbnail_", fileext = ".jpg")
+        received_path <- get_thumbnail(item, user_path)
+        received <- readBin(received_path, "raw", n = 8)
+        expected <- as.raw(c(0x4e, 0x41))
+        expect_equal(user_path, received_path)
+        expect_equal(received, expected)
+
+        # User-specified path (wrong extension)
+        user_path <- tempfile("thumbnail_", fileext = ".png")
+        received_path <- get_thumbnail(item, user_path)
+        received <- readBin(received_path, "raw", n = 8)
+        expected <- as.raw(c(0x4e, 0x41))
+        expect_equal(paste0(user_path, ".jpeg"), received_path)
+        expect_equal(received, expected)
```

Comment:
This results in `thumbnail_blah.png.jpeg` is that what we want?

Reply:
It seems reasonable to me. The converse (substituting rather than appending the correct extension) also seems reasonable.

This only occurs when the user has provided a file path with the incorrect extension. I could see an argument that in this case, it's better to preserve the entire original user-supplied path and just append the correct extension. That's what @nealrichardson did in his suggestion, so I assumed that it was done with that intent.

Reply:
Right. IMO I don't think it's worth trying to be even more clever with the filenames.